superset-frontend/src/features/alerts/components/NotificationMethod.tsx

```diff
@@ -257,51 +259,47 @@ export const NotificationMethod: FunctionComponent<NotificationMethodProps> = ({
   };
 
   useEffect(() => {
-    if (
-      method &&
-      [
-        NotificationMethodOption.Slack,
-        NotificationMethodOption.SlackV2,
-      ].includes(method) &&
-      !slackOptions[0]?.options.length
-    ) {
+    if (!slackOptions[0]?.options.length) {
       fetchSlackChannels({ types: ['public_channel', 'private_channel'] })
         .then(({ json }) => {
           const { result } = json;
-
           const options: SlackOptionsType = mapChannelsToOptions(result);
 
           setSlackOptions(options);
 
           if (isFeatureEnabled(FeatureFlag.AlertReportSlackV2)) {
-            // map existing ids to names for display
+            // for edit mode, map existing ids to names for display if slack v2
             // or names to ids if slack v1
             const [publicOptions, privateOptions] = options;
-
-            setSlackRecipients(
-              mapSlackValues({
-                method,
-                recipientValue,
-                slackOptions: [
-                  ...publicOptions.options,
-                  ...privateOptions.options,
-                ],
-              }),
-            );
-            if (method === NotificationMethodOption.Slack) {
-              onMethodChange({
-                label: NotificationMethodOption.Slack,
-                value: NotificationMethodOption.SlackV2,
-              });
+            if (
+              method &&
```

Comment:
This line shouldn't be necessary, right?

Reply:
It's to help typescript assert that the value exists, rather than runtime execution. e.g., for `NotificationMethodOption` as it doesn't include `undefined`. I could coerce the type, but I think that sometimes looks like a shortcut, too. wdyt?

Reply:
I'm going to merge, but if you suggest another way to approach this, I'll put it up in a new PR.